status.py

```diff
@@ -1,74 +1,114 @@
 #!/usr/bin/python3
 
 import asyncio
-from urllib.parse import urljoin
+import sys
+from email.utils import parsedate_to_datetime
 
-import aiohttp
-import yaml
+import httpx  # in python-httpx and python-h2 package
+import orjson  # in python-orjson package
+import ruamel.yaml  # in python-ruamel-yaml package
 
+SOURCE_YAML = "mirrors.yaml"
+OUTPUT_JSON = "list.json"
 
-def humantime(t: int) -> str:
-    """seconds -> XhYmZs"""
-    m, s = divmod(t, 60)
-    h, m = divmod(m, 60)
-    d, h = divmod(h, 24)
-    ret = ""
-    if d:
-        ret += "%dd" % d
-    if h:
-        ret += "%dh" % h
-    if m:
-        ret += "%dm" % m
-    if s:
-        ret += "%ds" % s
-    if not ret:
-        ret = "0s"
-    return ret
 
+FILES = (
+    "lastupdate",
+    "x86_64/archlinuxcn.db",
+    "any/archlinuxcn.db",
+)
 
-async def get_lastupdate(session, name, url):
+
+async def fetch(client, mirror, file):
+    provider = mirror["provider"]
+    url = mirror["url"]
     try:
-        res = await session.get(url)
-        content = await res.text()
-        dt = int(content)
-        print(f"{name} done.")
-        return name, dt
-    except Exception as e:
-        return name, e
+        response = await client.head(url + file)
+        if response.status_code == httpx.codes.OK:
+            timestamp = parsedate_to_datetime(response.headers.get("Last-Modified"))
+            return provider, file, timestamp, None
+        response.raise_for_status()
+    except (httpx.HTTPError, httpx.InvalidURL) as error:
+        return provider, file, None, error
 
 
-async def main():
-    with open("mirrors.yaml") as f:
-        data = yaml.load(f, Loader=yaml.FullLoader)
+async def gather(mirrors):
+    mirrors.append(
+        {
+            "provider": "tier0",
+            "url": "https://repo.archlinuxcn.org/",
+        }
+    )
+    client = httpx.AsyncClient(
+        headers={"User-Agent": "httpx/0.25.2"},
+        http2=True,
+        timeout=10,
+        follow_redirects=True,
```

Comment:
Without this parameter, the CERNET mirror site cannot be tested, although testing this mirror site is meaningless.